lib/job-simulation.rb

```diff
@@ -2,13 +2,43 @@
 require './Queue.rb'
 
 class JobSimulation
-  attr_reader :workers, :waiting, :roll
+  attr_reader :workers, :waiting, :roll,:count
 
-  def initialize (jobs_available, job_seekers)
+  def initialize(jobs_available, total_jobs)
+    #setting total numbers
+    @count = total_jobs
+    all = Array.new
+    (0..count).each{|number|all.push("Worker ##{number+1}")}
+    #splitting workers between those working and those waiting
+    starting_workers = all.slice(0..jobs_available)
+    waiting_workers = all.slice(jobs_available + 1..-1)
 
+    @workers = Stack.new
+    starting_workers.each{|worker|workers.push(worker)}
+
+    @waiting = Queue.new
+    waiting_workers.reverse.each{|worker|waiting.enqueue(worker)}
   end
 
   def cycle
+    roll = rand(1..6)
```

Comment:
careful: this is a local variable, not the same as the @roll up above in the attr_reader